test/Kestrel.Core.Tests/Http1ConnectionTests.cs

```diff
@@ -81,6 +81,37 @@ public void Dispose()
             _pipelineFactory.Dispose();
         }
 
+        [Fact]
+        public async Task TakeMessageHeadersSucceedsWhenHeaderValueContainsUTF8()
+        {
+            string headerName = "Header";
+            string headerValue = "Fran�ois";
+            string headerLine = $"{headerName}: {headerValue}\r\n";
+            _http1Connection.Reset();
+
+            await _application.Output.WriteAsync(Encoding.UTF8.GetBytes($"{headerLine}\r\n"));
+            var readableBuffer = (await _transport.Input.ReadAsync()).Buffer;
+
+            _http1Connection.TakeMessageHeaders(readableBuffer, out _consumed, out _examined);
+            _transport.Input.AdvanceTo(_consumed, _examined);
+
+            Assert.Equal(headerValue, _http1Connection.RequestHeaders[headerName]);
+        }
+
+        [Fact]
+        public async Task TakeMessageHeadersThrowsWhenHeaderValueContainsExtendedASCII()
+        {
+            string headerName = "Header";
+            string headerValue = "Fran�ois";
+            string headerLine = $"{headerName}: {headerValue}\r\n";
+            _http1Connection.Reset();
+
+            await _application.Output.WriteAsync(Encoding.GetEncoding("ISO-8859-1").GetBytes($"{headerLine}\r\n"));
+            var readableBuffer = (await _transport.Input.ReadAsync()).Buffer;
+
+            var exception = Assert.Throws<InvalidOperationException>(() => _http1Connection.TakeMessageHeaders(readableBuffer, out _consumed, out _examined));
```

Comment:
This line isn't throwing at all on Mac and Linux. 😕

Reply:
Turns out Visual Studio is encoding `ç` as `<E7>` in the source code which is not interpreted correctly on Unix. 
```
@@ -102,7 +102,7 @@ public void Dispose()
         public async Task TakeMessageHeadersThrowsWhenHeaderValueContainsExtendedASCII()
         {
             string headerName = "Header";
-            string headerValue = "Fran<E7>ois";
+            string headerValue = "François";
             string headerLine = $"{headerName}: {headerValue}\r\n";
             _http1Connection.Reset();
```
I've switched it to `ç` using VS Code. If this doesn't work on windows, I'll just use raw bytes so there's no ambiguity.